### PR TITLE
RD-1001 Update LDAP documentation

### PR DESCRIPTION
**Pull request by geokala**

### CODE REVIEW

File: content/working_with/manager/ldap-integration.md [hunk 1]
```diff
@@ -6,56 +6,177 @@ weight: 1450
 ---
 {{< param product_name >}} provides a user management mechanism, so you can define different users with different permissions, and upon login perform authentication and authorization to control the users’ access to resources. 
 
-The users can be either defined and managed in {{< param product_name >}} itself, or you can configure your Manager to integrate with an LDAP-based user-management system.
-You must select one of these options, as you cannot do both, and you must configure your manager accordingly upon installation or immediately afterwards, when no actions were performed on it yet.  
+The users can be either defined and managed in {{< param product_name >}} itself, or you can configure your Manager to integrate with an LDAP-based user-management system. You must select one of these options, as you cannot do both.
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:9]:
```suggestion
The users can be either defined and managed in {{< param product_name >}} itself, or you can configure your {{< param cfy_manager_name >}} to integrate with an LDAP-based user-management system. You must select one of these options, as you cannot do both.
```

File: content/working_with/manager/ldap-integration.md [hunk 2]
```diff
@@ -6,56 +6,177 @@ weight: 1450
...
+
+If LDAP integration is enabled, {{< param product_name >}} system role membership should be configured using [user-groups]({{< relref "cli/maint_cli/usergroups.md" >}}).
+
+If you enable LDAP with existing users already on the manager those users will continue to exist but cannot be used to log in unless they match an LDAP username.
+e.g. if a user jbloggs exists on the manager before ldap is enabled then after LDAP is enabled a login is made by LDAP user jbloggs, the LDAP user jbloggs will be shown as the creator of any entities (e.g. blueprints, secrets, deployments) that were created by the original jbloggs user.
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:14]:
```suggestion
e.g. if a user jbloggs exists on the manager before LDAP is enabled, then after LDAP is enabled a login is made by LDAP user jbloggs, the LDAP user jbloggs will be shown as the creator of any entities (e.g. blueprints, secrets, deployments) that were created by the original jbloggs user.
```

File: content/working_with/manager/ldap-integration.md [hunk 3]
```diff
@@ -6,56 +6,177 @@ weight: 1450
...
+
 {{% tip title="User Management Credentials" %}}
 You must have {{< param cfy_manager_name >}} administrator permissions to perform user-management related actions.
 {{% /tip %}}
 
-## Managing users by Integrating with an LDAP System
-If you choose to integrate with an external user-management system, make sure your manager is configured accordingly:
+## Basic integration
+In order to enable LDAP user integration you must know:
+* The LDAP server address, e.g. ldap://192.0.2.1 or ldaps://192.0.2.4
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:28]:
```suggestion

* The LDAP server address, e.g. ldap://192.0.2.1 or ldaps://192.0.2.4
```

File: content/working_with/manager/ldap-integration.md [hunk 4]
```diff
@@ -6,56 +6,177 @@ weight: 1450
...
+## Advanced integration
+
+Defaults for active directory and non-active directory setups are given below. If your settings differ, or you need nested group lookups, consult the command options below to determine the correct settings to supply.
+
+Examples given assume a user attempting to log in as 'jbloggs' and an ldap domain of 'local.example'.
+
+For details of variables (e.g. {username}), see below.
+
+For active directory:
+* Base DN for lookups is the domain, split on '.', as a series of dc LDAP entities- e.g. dc=local,dc=example
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:57]:
```suggestion

* Base DN for lookups is the domain, split on '.', as a series of dc LDAP entities- e.g. dc=local,dc=example
```

File: content/working_with/manager/ldap-integration.md [hunk 5]
```diff
@@ -6,56 +6,177 @@ weight: 1450
...
+* LDAP objects are expected to have the following attributes:
+  - uid (required)- The username of the user, e.g. jbloggs
+  - givenName (optional)- The given name of the user- this may be blank. If not blank, it will be stored as part of the user details on the manager.
+  - sn (optional)- The surname of the user- this may be blank. If not blank, it will be stored as part of the user details on the manager.
+  - email (optional)- The e-mail address of the user- this may be blank. If not blank, it will be stored as part of the user details on the manager.
+  - memberOf (required)- If the user is a member of no groups other than their primary group (usually Domain Users) then they will be effectively unusable with {{< param product_name >}}, so this should be populated with the group membership.
+* Groups are looked up by consulting the memberOf attribute on a user (or, with nested lookups, on the groups).
+
+For other directories (e.g. openldap):
+* Base DN for lookups is the domain, split on '.', as a series of dc LDAP entities- e.g. dc=local,dc=example
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:69]:
```suggestion

* Base DN for lookups is the domain, split on '.', as a series of dc LDAP entities- e.g. dc=local,dc=example
```

File: content/working_with/manager/ldap-integration.md [hunk 6]
```diff
@@ -6,56 +6,177 @@ weight: 1450
...
+  - memberOf (optional)- If groups are present on user objects using this attribute they must also be on group objects if using nested lookups.
+* Groups are looked up using either the memberOf attribute (if present), or the group membership lookup query above.
+
+To override the above options, please consult the ldap set command options below.
+
+**Variables**
+
+Some of the options can accept variables which will be substituted. Such variables are provided enclosed in braces, e.g. "{base_dn}".
+The available variables for each option will be listed in the options below, and all come from the following list:
+* {base_dn}- The base DN for all LDAP searches. By default this is the same as the domain DN.
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:87]:
```suggestion

* {base_dn}- The base DN for all LDAP searches. By default this is the same as the domain DN.
```

File: content/working_with/manager/ldap-integration.md [hunk 7]
```diff
@@ -6,56 +6,177 @@ weight: 1450
...
-So, userA is now assigned to dev_tenant twice – once as a Viewer and once as a User. Upon logging into this tenant, the permissions userA will have will be a sum of the permissions of the two roles. For more information regaring the user-roles, see [Managing Roles.]({{< relref "working_with/manager/roles-management.md" >}})
+For example, let’s say jbloggs is a member of two Groups in LDAP – “team_leaders”, and “devs”. The team_leaders group is associated in {{< param product_name >}} with the group “all_tenants_viewers”, which is assigned to all of the manager's tenants with the role “Viewer”. The “devs” group is associated in {{< param product_name >}} with the group “dev_users”, which is assigned to dev_tenant with the role “User”.
+So, jbloggs is now assigned to dev_tenant twice – once as a Viewer and once as a User. Upon logging into this tenant, the permissions jbloggs will have will be a sum of the permissions of the two roles. For more information regaring the user-roles, see [Managing Roles.]({{< relref "working_with/manager/roles-management.md" >}})
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:179]:
```suggestion

So, jbloggs is now assigned to dev_tenant twice – once as a Viewer and once as a User. Upon logging into this tenant, the permissions jbloggs will have will be a sum of the permissions of the two roles. For more information regaring the user-roles, see [Managing Roles.]({{< relref "working_with/manager/roles-management.md" >}})
```

File: content/working_with/manager/ldap-integration.md [hunk 8]
```diff
@@ -6,56 +6,177 @@ weight: 1450
...
+* {object_dn}- The DN of the object (user or group) that is being checked for membership of groups.
+* {username}- The username of the user attempting to bind (for bind format) or the user attempting to log on (for all lookups).
+* {domain}- The raw domain, e.g. local.example.
 
 **Usage**
 
-```cfy ldap set [OPTIONS]```
+`cfy ldap set [OPTIONS]`
 
 **Options**
```
Comment (ofercloudify) [inline content/working_with/manager/ldap-integration.md:97]: The layout of this section is broken. add bullets/newlines as it all appears in one bulk.

**Review by ofercloudify — CHANGES_REQUESTED** [review]

File: content/working_with/manager/ldap-integration.md (hunk 1 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md:9]: Manager references updated in upcoming commit

File: content/working_with/manager/ldap-integration.md (hunk 3 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md:28]: LDAP uppercasing done where relevant in upcoming commit.

File: content/working_with/manager/ldap-integration.md (hunk 2 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md]: LDAP uppercased where appropriate.

File: content/working_with/manager/ldap-integration.md (hunk 4 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md:58]: Spacing updated in new commit.

File: content/working_with/manager/ldap-integration.md (hunk 5 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md:71]: Spacing updated in new commit.

File: content/working_with/manager/ldap-integration.md (hunk 6 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md:90]: Spacing updated in new commit.

File: content/working_with/manager/ldap-integration.md (hunk 7 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md:183]: Spacing updated in new commit.

File: content/working_with/manager/ldap-integration.md (hunk 8 above)
Comment (geokala) [reply content/working_with/manager/ldap-integration.md:100]: I've updated that section to valid markdown and tidied (or maybe messed!) it up a bit.

**Review by ofercloudify — APPROVED** [review]

**Pull request merged**